### ISSUE TITLE
Problems retrieving data with EmbeddedModelField type fields

### ISSUE THREAD

**Issue by rudolfce** [nesdis/djongo#15]
Hello

Since version 1.2.4 I've been having trouble using EmbeddedModelFields.

When I try to retrieve entries from the database, I get an error in Django. I was able to reproduce with a simpler code to try and isolate the problem, but I can't figure out how to fix it.

Example model:
```python
class Embedded(models.Model):
     text = models.CharField(max_length=100)
     class Meta:
         abstract = True

 class Test(models.Model):
     name = models.CharField(max_length=100)
     internal = models.EmbeddedModelField(model_container = Embedded)
```
So, from the django shell, I issue:
```python
Teste.objects.all()
```

I get the following error:

```
...
~/Documents/projetos/sheeter/env/lib/python3.6/site-packages/django/db/models/sql/compiler.py in results_iter(self, results)
    840             for row in rows:
    841                 if converters:
--> 842                     row = self.apply_converters(row, converters)
    843                 yield row
    844 

~/Documents/projetos/sheeter/env/lib/python3.6/site-packages/django/db/models/sql/compiler.py in apply_converters(self, row, converters)
    823         row = list(row)
    824         for pos, (convs, expression) in converters.items():
--> 825             value = row[pos]
    826             for converter in convs:
    827                 value = converter(value, expression, self.connection, self.query.context)

IndexError: list index out of range
```
Doing some debugging in the django compiler, I noticed it has to do with the id field. The converters consider id as a field, while row does not have it (hence, there is a 1 difference in length). If I include an _id field in the model, like I did in my Layout model in [this file](https://github.com/rudolfce/sheeter/blob/master/layouts/models.py), I get an error that looks like this:

```
...

  File "/home/rudolf/Documents/projetos/sheeter/env/lib/python3.6/site-packages/django/db/models/query.py", line 250, in __iter__
    self._fetch_all()
  File "/home/rudolf/Documents/projetos/sheeter/env/lib/python3.6/site-packages/django/db/models/query.py", line 1118, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/home/rudolf/Documents/projetos/sheeter/env/lib/python3.6/site-packages/django/db/models/query.py", line 63, in __iter__
    obj = model_cls.from_db(db, init_list, row[model_fields_start:model_fields_end])
  File "/home/rudolf/Documents/projetos/sheeter/env/lib/python3.6/site-packages/django/db/models/base.py", line 582, in from_db
    values = [values.pop() if f.attname in field_names else DEFERRED for f in cls._meta.concrete_fields]
  File "/home/rudolf/Documents/projetos/sheeter/env/lib/python3.6/site-packages/django/db/models/base.py", line 582, in <listcomp>
    values = [values.pop() if f.attname in field_names else DEFERRED for f in cls._meta.concrete_fields]
IndexError: pop from empty list
```
Again, the _id will be the problem: values won't have it, but cls._meta will consider it.

Version logging:
Occurs on 1.2.4 and 1.2.5;
Doesn't happen on 1.2.3.

**Comment by nesdis** [comment]
Hi Rudolf

How have you been!

Thanks for that analysis. Let me take a look at it. 
Did quite a bit of refactoring of code in 1.2.4 vs 1.2.3,

Structured the code to make it more readable. May have caused some problems. I will check asap.

Thanks

**Comment by rudolfce** [comment]
Hey! I'm fine, and you?

Nice to hear it. Please let me know if there's any extra information I can help with.

**Comment by nesdis** [comment]
Hi 

I am doing good, thank you!

Have you applied migrations after switching to new version. `manage.py migrate`? Also can you tell me if the database actually has the 'internal' embedded field in it, or it its just an empty collection?.

**Comment by nesdis** [comment]
Can you check and let me know if you issue is fixed with the code in (pull request)(https://github.com/nesdis/djongo/pull/16)

**Comment by rudolfce** [comment]
Hey there

I tried reseting the database (db.dropDatabase() on the db in question and then issuing a migration) only to be sure. On my experiment, I didn't try using empty fields - that might be also something worth trying out.

With the branch from the pull request, I can retrieve data normally - both for all entries and single entry. I can also update entries normally. If I swtich back to the pypi's version, I get the same error back. The fix seems to work for this bug.

**Comment by nesdis** [comment]
Great will release the next version with this fix.

**Comment by nesdis** [comment]
I have released a new version on pypi: 1.2.6. Your issue should be fixed in that.

**Comment by rudolfce** [comment]
Awsome! Thank you
